ecp5/pack.cc

```diff
@@ -3007,11 +3008,13 @@ void Arch::assignArchInfo()
 
             // Output register mode (REGMODE_{A,B}). Valid options are 'NOREG' and 'OUTREG'.
             std::string regmode_a = str_or_default(ci->params, id("REGMODE_A"), "NOREG");
-            if (!(regmode_a == "NOREG" || regmode_a == "OUTREG"))
-                NPNR_ASSERT_FALSE_STR("bad DP16KD REGMODE_A configuration '" + regmode_a + "'");
+            if (regmode_a != "NOREG" && regmode_a != "OUTREG")
+                log_error("DP16KD %s has invalid REGMODE_A configuration '%s'\n", ci->name.c_str(ctx),
```

Comment:
`IdString`s are associated with `BaseCtx`, the base class of Arch, so `c_str(this)` should suffice